src/components/templates/Auditorium/components/AllSectionPreviews/AllSectionPreviews.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useMemo, useCallback } from "react";
+import classNames from "classnames";
+import { useHistory } from "react-router-dom";
+import { sample } from "lodash";
+
+import { AuditoriumEmptyBlocksCount } from "types/auditorium";
+import { AuditoriumVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { chooseAuditoriumSize } from "utils/auditorium";
+import { enterVenue } from "utils/url";
+
+import { useAuditoriumSections } from "hooks/auditoriumSections";
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+
+import { BackButton } from "components/atoms/BackButton";
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { IFrame } from "components/atoms/IFrame";
+
+import { SectionPreview } from "../SectionPreview";
+
+import "./AllSectionPreviews.scss";
+
+export interface SectionPreviewsProps {
+  venue: WithId<AuditoriumVenue>;
+}
+
+export const AllSectionPreviews: React.FC<SectionPreviewsProps> = ({
+  venue,
+}) => {
+  const { push: openUrlUsingRouter } = useHistory();
+
+  const { parentVenue } = useRelatedVenues({
+    currentVenueId: venue.id,
+  });
+  const parentVenueId = parentVenue?.id;
+
+  const {
+    auditoriumSections,
+    toggleFullAuditoriums,
+    isFullAuditoriumsHidden,
+    enterSection,
+    notFullSections,
+  } = useAuditoriumSections(venue);
+
+  const sectionsCount = auditoriumSections.length;
+
+  const auditoriumSize = chooseAuditoriumSize(sectionsCount);
+
+  const sectionPreviews = useMemo(
+    () =>
+      auditoriumSections.map((section) => (
+        <SectionPreview
+          key={section.id}
+          section={section}
+          venue={venue}
+          enterSection={enterSection}
+        />
+      )),
+    [auditoriumSections, venue, enterSection]
+  );
+
+  const emptyBlocks = useMemo(
+    () =>
+      Array(AuditoriumEmptyBlocksCount[auditoriumSize])
+        .fill(0)
+        .map((_, index) => (
+          <div
+            key={index}
+            className={`AllSectionPreviews__empty-block--${index + 1}`}
+          />
+        )),
+    [auditoriumSize]
+  );
+
+  const notFullSectionIds = useMemo(
```

Comment:
Question. Probably there is a better name for this

Reply:
Changed in https://github.com/sparkletown/sparkle/pull/1251/commits/5bbf188eadabe69e3ca5046a8b4e3a8d09d1535e